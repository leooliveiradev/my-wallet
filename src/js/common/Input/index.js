import styled from 'styled-components';
import { fromTheme } from 'app/js/utils';

const Input = styled.input.attrs({
  type: ({ type }) => type || 'text',
})`
  width: 26rem;
  height: 2.6rem;
  padding: 0 1.2rem;
  background-color: ${fromTheme('color.secondary')};
  border-radius: ${({ rounded }) => (rounded ? '20px' : '0px')};
  border: solid 1px ${fromTheme('color.lightGray')};
  font-family: ${fromTheme('fontStyle.primary')};
  font-size: 1rem;
  color: ${fromTheme('color.lightGray3')};
  outline: none;
  &::placeholder {
    font-size: 1rem;
    text-transform: capitalize;
    color: ${fromTheme('color.lightGray')};
  }
`;

export default Input;
