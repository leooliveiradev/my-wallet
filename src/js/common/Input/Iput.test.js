import React from 'react';
import Input from 'app/js/common/Input';
import renderer from 'react-test-renderer';

describe('Input snapshot', () => {
  it('should render the Input', () => {
    const component = renderer.create(
      <Input
        type="text"
        onChange={event => console.log(event.target.value)}
        name="transaction"
        placeholder="Transaction"
        rounded
      />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
