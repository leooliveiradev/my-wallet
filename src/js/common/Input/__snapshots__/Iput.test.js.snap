// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot should render the Input 1`] = `
<input
  className="sc-bdVaJa iiTjkL"
  name="transaction"
  onChange={[Function]}
  placeholder="Transaction"
  type="text"
/>
`;
